File: frontend/services/policyService.ts
import { http } from "@/lib/http";
import type {
  ApiResponse,
  CreatePolicyRequest,
  PagedResponse,
  PageParams,
  PolicyResponse,
} from "@/lib/types";

const unwrap = <T>(p: Promise<{ data: ApiResponse<T> }>) => p.then((r) => r.data.data);

export const policyService = {
  list: (params: PageParams = {}) =>
    unwrap(http.get<ApiResponse<PagedResponse<PolicyResponse>>>("/policies", { params })),
  /** All policies (unpaged) — for the policy picker on the new-claim form. */
  options: () => unwrap(http.get<ApiResponse<PolicyResponse[]>>("/policies/options")),
  get: (id: number) => unwrap(http.get<ApiResponse<PolicyResponse>>(`/policies/${id}`)),
  create: (body: CreatePolicyRequest) =>
    unwrap(http.post<ApiResponse<PolicyResponse>>("/policies", body)),
  cancel: (id: number) =>
    unwrap(http.post<ApiResponse<PolicyResponse>>(`/policies/${id}/cancel`, {})),
};

File: backend/src/main/java/com/niyotechnologies/claimlens/policy/service/PolicyService.java
package com.niyotechnologies.claimlens.policy.service;

import com.niyotechnologies.claimlens.policy.dto.request.CreatePolicyRequest;
import com.niyotechnologies.claimlens.policy.dto.response.PolicyResponse;
import com.niyotechnologies.claimlens.common.response.PagedResponse;

import java.util.List;

public interface PolicyService {

    PolicyResponse createPolicy(CreatePolicyRequest request);

    PolicyResponse cancelPolicy(Long policyId);

    PolicyResponse getPolicy(Long policyId);

    /** Paged directory listing for the Policies screen. */
    PagedResponse<PolicyResponse> getPolicies(int page, int size);

    /** All policies (unpaged) — for the policy picker on the new-claim form. */
    List<PolicyResponse> getPolicyOptions();
}
